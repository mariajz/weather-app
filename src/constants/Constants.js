export const degreeSymbol = '\u00B0';
