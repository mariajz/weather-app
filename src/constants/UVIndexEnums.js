export const UVWarnings = {
    low: 'Minimal risk of harm. Enjoy outdoor activities without special precautions.',
    moderate: 'Some risk of harm. Wear sunglasses and use sunscreen.',
    high: 'Avoid outdoor activities during peak sun hours.',
    veryHigh:
        'Very high risk of harm. Seek shade and use broad-spectrum sunscreen.',
    extreme: 'Stay indoors. If not, use maximum sun protection.',
    default: '',
};
