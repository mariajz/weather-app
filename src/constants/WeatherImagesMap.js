import { images } from '../../assets/images';
const {
    PartlyCloudy,
    ModerateRain,
    PatchyRainPossible,
    Sunny,
    Clear,
    Overcast,
    Cloudy,
    LightRain,
    ModerateRainAtTimes,
    HeavyRain,
    HeavyRainAtTimes,
    ModerateOrHeavyFreezingRain,
    ModerateOrHeavyRainShower,
    ModerateOrHeavyRainWithThunder,
    Mist,
    other,
} = images;

export const weatherImages = {
    'Partly cloudy': PartlyCloudy,
    'Moderate rain': ModerateRain,
    'Patchy rain possible': PatchyRainPossible,
    Sunny: Sunny,
    Clear: Clear,
    Overcast: Overcast,
    Cloudy: Cloudy,
    'Light rain': LightRain,
    'Light drizzle': LightRain,
    'Moderate rain at times': ModerateRainAtTimes,
    'Heavy rain': HeavyRain,
    'Heavy rain at times': HeavyRainAtTimes,
    'Moderate or heavy freezing rain': ModerateOrHeavyFreezingRain,
    'Moderate or heavy rain shower': ModerateOrHeavyRainShower,
    'Moderate or heavy rain with thunder': ModerateOrHeavyRainWithThunder,
    'Patchy light drizzle': ModerateRainAtTimes,
    'Patchy light rain': LightRain,
    'Light rain shower': LightRain,
    Mist: Mist,
    Fog: Mist,
    other: other,
};
