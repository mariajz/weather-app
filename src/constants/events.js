export const events = {
    SHOW_LOADER: 'SHOW_LOADER',
    HIDE_LOADER: 'HIDE_LOADER',
    SHOW_POPUP: 'SHOW_POPUP',
    REMOVE_POPUP: 'REMOVE_POPUP',
};
