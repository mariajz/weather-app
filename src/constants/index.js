export { weatherImages } from './WeatherImagesMap';
export { UVWarnings } from './UVIndexEnums';
export { events } from './events';
