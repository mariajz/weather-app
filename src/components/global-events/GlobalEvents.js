import React, { useEffect, useState } from 'react';
import Emitter from 'tiny-emitter/instance';
import { events } from '../../constants';
import Loader from '../loader';
import Popup from '../popup';

const GlobalEvents = () => {
    const [showLoader, setShowLoader] = useState(false);
    const [showPopup, setShowPopup] = useState(false);
    const [popup, setPopup] = useState({});

    useEffect(() => {
        Emitter.on(events.SHOW_LOADER, () => {
            setShowLoader(true);
        });
        Emitter.on(events.HIDE_LOADER, () => {
            setShowLoader(false);
        });
        Emitter.on(events.SHOW_POPUP, popupProps => {
            setShowPopup(true);
            setPopup(popupProps);
        });
        Emitter.on(events.REMOVE_POPUP, () => {
            setShowPopup(false);
        });
    }, []);

    return (
        <>
            <If condition={showPopup}>
                <Popup popupProps={popup} testID="popup" />
            </If>
            <If condition={showLoader}>
                <Loader testID="loader" />
            </If>
        </>
    );
};

export default GlobalEvents;
