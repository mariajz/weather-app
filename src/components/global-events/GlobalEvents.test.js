import { render } from '@testing-library/react-native';
import React from 'react';
import { View as MockView } from 'react-native';
import GlobalEvents from './GlobalEvents';
import Emitter from 'tiny-emitter/instance';
import { events } from '../../constants';

jest.mock('../popup', () => props => <MockView {...props} />);
jest.mock('../loader', () => props => <MockView {...props} />);

describe('GlobalEvents', () => {
    it('should render null if no events are emitted', () => {
        const container = render(<GlobalEvents />);

        expect(container).toMatchSnapshot();
    });

    it('should render loader when SHOW_LOADER is emitted', () => {
        const { getByTestId } = render(<GlobalEvents />);

        Emitter.emit(events.SHOW_LOADER, {});

        expect(getByTestId('loader')).toBeDefined();
    });

    it('should remove loader when HIDE_LOADER is emitted', () => {
        const { queryByTestId } = render(<GlobalEvents />);

        Emitter.emit(events.HIDE_LOADER, {});

        expect(queryByTestId('loader')).toBeNull();
    });

    it('should render popup when SHOW_POPUP is emitted', () => {
        const { getByTestId } = render(<GlobalEvents />);

        Emitter.emit(events.SHOW_POPUP, {});

        expect(getByTestId('popup')).toBeDefined();
    });

    it('should remove popup when REMOVE_POPUP is emitted', () => {
        const { queryByTestId } = render(<GlobalEvents />);

        Emitter.emit(events.REMOVE_POPUP, {});

        expect(queryByTestId('popup')).toBeNull();
    });
});
