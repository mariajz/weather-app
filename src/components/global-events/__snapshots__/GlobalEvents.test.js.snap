// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalEvents should render null if no events are emitted 1`] = `null`;
